.dockerignore

```diff
@@ -0,0 +1,9 @@
+# Ignore everything
+**
+
+!setup.py
+!README.md
+!optuna
+!examples
```

Comment:
How about whitelisting unit tests and docs related directories as well? I believe the provided Dockerfile is used by developers as well as users.

Reply:
I think we would mount the project directory to the container working directory during development.

Reply:
```
# fix optuna code
# fix test

# Run the test using the fixed code
docker run ... -v $(pwd):/workspaces <imane> pytest tests/...
```

Reply:
This VSCode doc might be helpful: https://code.visualstudio.com/docs/remote/containers

Reply:
I have excluded `docs` and `tests` because:
- We don't need them when running a python script that uses optuna.
- During development, we usually mount the project directory so the dev image doesn't need to contain them.

Reply:
Thanks for the detailed explanation. Mounting a completely separate directory when developing makes sense.